ironfish/src/network/peerNetwork.ts

```diff
@@ -1361,15 +1358,11 @@ export class PeerNetwork {
       return
     }
 
-    if (this.node.memPool.acceptTransaction(transaction)) {
-      this.onTransactionAccepted.emit(transaction, received)
-    }
-
-    // Check 'exists' rather than 'accepted' to allow for rebroadcasting to nodes that
-    // may not have seen the transaction yet
-    if (this.node.memPool.exists(transaction.hash())) {
-      this.broadcastTransaction(transaction)
-    }
+    // This transaction is valid and has not yet been seen so sync it to the
+    // mempool, gossip to all peers and emit telemetry event for new transaction seen
+    this.node.memPool.acceptTransaction(transaction)
```

Comment:
Doing an early return now so at this point the transaction has not already been seen in the mempool. We've also confirmed that it is valid. If it hasn't been seen in the mempool we always want to

1. gossip
2. accept into the mempool
3. sync to the wallet
4. submit telemetry event that a new transaction was seen